doc/website/blog/2021-11-1-sync-objects-in-external-database.md

```diff
@@ -0,0 +1,305 @@
+---
+title: Ent Hooks - Synchronize Objects In External Databases
+author: Ariel Mashraki
+authorURL: https://github.com/a8m
+authorImageURL: "https://avatars0.githubusercontent.com/u/7413593"
+authorTwitter: arielmashraki
+image: https://entgo.io/images/assets/sqlcomment/share.png
+---
+
+One of the common questions we get from the Ent community is how to synchronize objects or references between the
+database backing an Ent application (e.g. MySQL or PostgreSQL) with external services. For example, users would like
+to create or delete a record from in their CRM when a user is created or deleted in Ent, publish a message to a
+[Pub/Sub system](https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern) when an entity is updated, or verify
+references to blobs in object storages such as AWS S3 or Google Cloud Storage.
+
+Ensuring consistency between two separate data systems is not a simple task.  When we want to propagate, for example,
+deletion of a record in one system to another, there is no obvious way to guarantee that the two systems will end in a
+synchronized state, since one of them may fail and the network link between them may be slow or down. Having said that,
+and especially with the prominence of microservices-architectures these problems have become more common, and
+distributed systems researchers have come up with patterns to solve them, such as the
+[Saga Pattern](https://microservices.io/patterns/data/saga.html).
+
+The application of these patterns is usually complex and difficult, and so in many cases architects do not go after a
```

Comment:
`a "perfect"` what? Or maybe `the "perfect"`

Reply:
Should a "perfect" design or "perfect" system